### PR TITLE
Refactor APIs/Interfaces and fix tests

### PR DESCRIPTION
**Pull request by sedroche**
- Update the APIs to match the spec
- Simplify the params
- fix tests

### CODE REVIEW

File: pkg/deploys/fuse/deployer.go [hunk 1]
```diff
@@ -20,122 +21,128 @@ import (
...
+	k8sClient kubernetes.Interface
+	osClient  *openshift.ClientFactory
 }
 
-func NewDeployer(id string) *FuseDeployer {
-	return &FuseDeployer{id: id}
-}
+func NewDeployer() *FuseDeployer {
+	// Instantiate loader for kubeconfig file.
+	kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
```
Comment (sedroche) [inline pkg/deploys/fuse/deployer.go]: I have moved these here as they were being passed from the server but only used here in the fuse deployer. I am not sure if I should create objects like this though? i.e. panic in a "constructor". I wanted to keep a parameter-less "constructor" for each `Deployer` though

Comment (maleck13) [reply pkg/deploys/fuse/deployer.go]: Not a fan of this change. It makes it very difficult to unit test for instance. The kubeconfig here has to be loaded from disk and will create a real client targeting a real cluster only.
So i think having the dependency injected is much clearer from an API and testing perspective. What is the reason for wanting an empty constructor?

File: cmd/broker/main.go [hunk 2]
```diff
@@ -62,29 +62,16 @@ func runWithContext(ctx context.Context) error {
...
-	)
-	cfg, err := kubeconfig.ClientConfig()
-	if err != nil {
-		return errors.Wrap(err, "error creating kube client config")
-	}
-	k8sClient := k8sclient.GetKubeClient()
-	osClient := openshift.NewClientFactory(cfg)
-
-	ctrlr := controller.CreateController(namespace, k8sClient, osClient)
+	ctrlr := controller.CreateController([]controller.Deployer{
```
Comment (maleck13) [inline cmd/broker/main.go]: I think it is fine to inject dependencies here

File: pkg/deploys/fuse/deployer.go (hunk 1 above)
Comment (sedroche) [reply pkg/deploys/fuse/deployer.go]: Yeah, it felt wrong and your concerns make sense. I'll revert back to passing in params.
I wanted an empty constructor as I wanted each deployer to be responsible for their own setup. A minor issue though and as I said ill revert

**Comment by maleck13** [comment]
Is this still relevant?

**Comment by sedroche** [comment]
@maleck13 We decided to wait on this as we were close to a release. The work is still valid imo I just need to do a rebase

**Pull request closed without merging**